pex/resolver.py

```diff
@@ -1310,7 +1310,7 @@ def resolve_from_pex(
     )
     installed_distributions = OrderedSet()  # type: OrderedSet[InstalledDistribution]
     for target in unique_targets:
-        pex_env = PEXEnvironment(pex, target=target)
+        pex_env = PEXEnvironment.mount(pex, target=target)
```

Comment:
Totally unrelated to this change, but: it occurs to me that safely cleaning up the PEX_ROOT would probably require PEX's cooperation... a process like this isn't holding any sort of lock that would prevent someone from deleting this directory.

Reply:
Unless you know I trick I don't - you're butting up against Posix again and - even Linux. No clue about macOS facilities for non-cooperative directory locking. I'm not even sure after re-mounting Linux fs -mand that this applies to directories.